File: frontend/src/api/coverLetter.ts
import client from './client'

export interface CoverLetter {
  id: string
  user_id: string
  resume_id: string
  job_id?: string
  company_name: string
  hiring_manager?: string
  content?: string
  file_path_pdf?: string
  file_path_docx?: string
  created_at: string
  updated_at: string
}

export const coverLetterApi = {
  generate: (resume_id: string, jd_text: string, company_name: string, api_key: string, hiring_manager?: string, job_id?: string) =>
    client.post<CoverLetter>('/cover-letter/generate', { resume_id, jd_text, company_name, api_key, hiring_manager, job_id }),
  list: () => client.get<CoverLetter[]>('/cover-letter/'),
  get: (id: string) => client.get<CoverLetter>(`/cover-letter/${id}`),
  downloadPdf: (id: string) => client.get(`/cover-letter/${id}/download/pdf`, { responseType: 'blob' }),
  delete: (id: string) => client.delete(`/cover-letter/${id}`),
}

File: backend/app/schemas/cover_letter.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class CoverLetterGenerateRequest(BaseModel):
    resume_id: str
    jd_text: str = Field(..., min_length=10)
    company_name: str
    hiring_manager: Optional[str] = None
    job_id: Optional[str] = None
    api_key: str


class CoverLetterResponse(BaseModel):
    id: str
    user_id: str
    resume_id: str
    job_id: Optional[str]
    company_name: str
    hiring_manager: Optional[str]
    content: Optional[str]
    file_path_pdf: Optional[str]
    file_path_docx: Optional[str]
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
